app/components/funding_widget_component.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class FundingWidgetComponent < ViewComponent::Base
+  attr_reader :path, :form
+
+  def initialize(form, path)
+    @form = form
+    @path = path
+  end
+
+  def grouped_options
+    {}.tap do |options|
```

Comment:
Found this quite hard to follow... not sure if this is any better/if its right, but:

```
form.subjects.index_by { |_, v| v[:education] }.transform_values { |v| v.first }
```

Reply:
I've refactored this into an array and extracted it to smaller methods to make it simpler to understand.